### PR TITLE
[DM-3311] LWM2M agent process only one of a multiple Resource Instances sent in a row

### PR DESCRIPTION
**Pull request by bggov**
…

### CODE REVIEW

File: content/change-logs/ui-c8y-1019-6-13-sent-in-a-row-lwm2m-resource-instance-not-processed.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+date: ""
+title: LWM2M agent process only one of a multiple Resource Instances sent in a row
```
Comment (BeateRixen) [inline content/change-logs/ui-c8y-1019-6-13-sent-in-a-row-lwm2m-resource-instance-not-processed.md:3]:
```suggestion
title: LWM2M agent processes all multiple resource instance IDs correctly
```

File: content/change-logs/ui-c8y-1019-6-13-sent-in-a-row-lwm2m-resource-instance-not-processed.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+component:
+  - value: component-1KLUzmqfe
+    label: LWM2M
+build_artifact:
+  - value: tc-pjJiURv9Y
+    label: ui-c8y
+ticket: DM-3311
+version: 1019.6.13
+---
+When a LWM2M device sends a composite request with multiple Resource Instance IDs from a same Resource ID and this Resource ID has an additional action defined, only one of these Resource Instances values was sent to this additional action for processing. The same behavior is observed when you try to do a composite read (`cread`) device operation with a row of Resource Instance IDs. These issues are resolved and all requested Resource Instance IDs are processed
```
Comment (BeateRixen) [inline content/change-logs/ui-c8y-1019-6-13-sent-in-a-row-lwm2m-resource-instance-not-processed.md:17]:
```suggestion
When a LWM2M device sends a composite request with multiple resource instance IDs from the same resource ID and this resource ID has an additional action defined, only one of these resource instance values was sent to this additional action for processing. The same behavior was observed when you try to do a composite read (`cread`) device operation with a row of resource instance IDs. These issues are resolved and all requested resource instance IDs are now processed.
```

**Review by BeateRixen — APPROVED** [review]

**Review by Oezge-Akin — COMMENTED** [review]
Could you rename the filename? cumulocity-1018.540.66-... and not ui-c8y...
The file should be in change-logs/device-management directory.

File: content/change-logs/ui-c8y-1019-6-13-sent-in-a-row-lwm2m-resource-instance-not-processed.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  - value: change-2c7RdTdXo4
+    label: Fix
+component:
+  - value: component-1KLUzmqfe
+    label: LWM2M
+build_artifact:
+  - value: tc-pjJiURv9Y
+    label: ui-c8y
+ticket: DM-3311
+version: 1019.6.13
```
Comment (Oezge-Akin) [inline content/change-logs/ui-c8y-1019-6-13-sent-in-a-row-lwm2m-resource-instance-not-processed.md:15]:
```suggestion
version: 1020.308.0
```

**Comment by Oezge-Akin** [comment]
> DO NOT MERGE! This will be merged when https://github.softwareag.com/IOTA/c8y-lwm2m/pull/26 is merged to develop and we have a real build `version`

CD changes are merged when it's in eu-latest (Zone 1).
So we need to keep this open for a while

File: content/change-logs/ui-c8y-1019-6-13-sent-in-a-row-lwm2m-resource-instance-not-processed.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  - value: change-2c7RdTdXo4
+    label: Improvement
+component:
+  - value: component-1KLUzmqfe
+    label: LWM2M
+build_artifact:
+  - value: tc-pjJiURv9Y
+    label: ui-c8y
+ticket: DM-3311
+version: 1019.6.13
```
[the remark this answers is not in the document]
Comment (Oezge-Akin) [reply content/change-logs/ui-c8y-1019-6-13-sent-in-a-row-lwm2m-resource-instance-not-processed.md:15]: now we have the version, added the suggestion below. resolving this thread.

File: content/change-logs/device-management/cumulocity-10-20-TBD-0-sent-in-a-row-lwm2m-resource-instance-not-processed.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  - value: change-VSkj2iV9m
+    label: Fix
+component:
+  - value: component-1KLUzmqfe
+    label: LWM2M
+build_artifact:
+  - value: tc-QHwMfWtBk7
+    label: cumulocity
+ticket: DM-3311
+version: 10.20.TBD.0
```
Comment (Oezge-Akin) [inline content/change-logs/device-management/cumulocity-10-20-TBD-0-sent-in-a-row-lwm2m-resource-instance-not-processed.md:15]:
```suggestion
version: 10.20.308.0
```

**Review by Oezge-Akin — APPROVED** [review]

**Pull request merged**